vendor_data.py

```diff
@@ -1,241 +1,375 @@
             ProjectName = pyip.inputStr("What is the project name? ")
             SourceLang = pyip.inputStr("What is the source language? ")
             VendorName = pyip.inputStr("What's the vendor's name? ")
             TargLang = pyip.inputStr("Into which language will the vendor translate? ")
-            NewWordRate = pyip.inputNum("What is the vendor's word rate in EUR? Should be between 0.01 and 0.15. If higher, choose another vendor. ", blank = True)
-            VendorWordRate = NewWordRate if NewWordRate else WordRate
-            NewPreferred = pyip.inputBool("True or False: Is this vendor a preferred vendor? If neither, leave blank. ", blank = True, default=None)
-            Preferred = NewPreferred if NewPreferred else Preferred
             NewVendorMail = pyip.inputStr("What is the vendor's email address? ", blank = True, default="")
-            VendorMail = NewVendorMail if NewVendorMail else VendorMail
-            NewCatTool= pyip.inputMenu(["XTM", "Trados Studio", "MemoQ", "Memsource"], prompt = "In which tool will the vendor be working?", blank = True, default = "XTM")
+            if NewVendorMail:
+                CheckVendorMail(NewVendorMail)
+                VendorMail = NewVendorMail
+            else:
+                VendorMail = VendorMail
+            NewWordRate = pyip.inputFloat("What is the vendor's word rate in EUR? Should be between 0.01 and 0.15. If higher, choose another vendor. ", blank = True)
+            if NewWordRate:
+                CheckWordRate(NewWordRate)
+                VendorWordRate = NewWordRate
+            else:
+                WordRate = WordRate
+            NewCatTool= pyip.inputMenu(VendorData.CatTools, prompt = "In which tool will the vendor be working?", blank = True, default = "XTM")
             CatTool = NewCatTool if NewCatTool else CatTool
-            done = pyip.inputStr("Are you done? ")
-            if done.lower() == "yes":
-                Vndr=VendorData(ProjectName, SourceLang, VendorName, TargLang, WordRate, Preferred, VendorMail, CatTool)
-                Excel = pyip.inputStr("Do you want add this vendor to the excel file? ")
-                if Excel.lower() == "yes":
-                    Vndr.to_excel()
-    if args.modify == True:
+            NewPreferred = pyip.inputBool("True or False: Is this vendor a preferred vendor? If neither, leave blank. ", blank = True, default=None)
+            Preferred = NewPreferred if NewPreferred else Preferred
+            done = pyip.inputStr("Are you done? ") #user gets the chance to change value for done to "no"
+            
+        Vndr=VendorData(ProjectName, SourceLang, TargLang, VendorName, VendorMail, WordRate, CatTool, Preferred)
+        Excel = pyip.inputStr("Do you want to add this vendor to the excel file? ") #user gets the chance to write data to excel
+        if Excel.lower() == "yes":
+            Vndr.ToExcel() #data written to excel

```

Comment:
I would add something like:

```python
    print("Vendor was added.")
else:
    print("Action cancelled.")
```

So the user has confirmation of what happened